server/test/smart-scheduling.test.ts

```diff
@@ -70,23 +99,144 @@ describe("GET /smart-scheduling/schedules/states/:state.ndjson", () => {
     const data = ndjsonParse(res.body);
     expect(data).toHaveLength(1);
   });
+
+  describe("`has-availability` extension", () => {
+    const extension =
+      "http://fhir-registry.smarthealthit.org/StructureDefinition/has-availability";
+
+    // Takes `LocationAvailability.available` value and
+    // expected `has-availability` extension value in SMART SL.
+    it.each([
+      [Availability.YES, "some"],
+      [Availability.NO, "none"],
+      [Availability.UNKNOWN, "unknown"],
+      // Test no availability record at all.
+      [null, "unknown"],
+    ])("represents %j as %j", async (available, expected) => {
+      const location = await createRandomLocation({
+        availability: available && { available },
+      });
+      const response = await context.client<any[]>(
+        `smart-scheduling/schedules/states/${location.state}.ndjson`,
+        { parseJson: ndjsonParse }
+      );
```

Comment:
Realized I could use the `parseJson` option to make it handle ND-JSON, which is neat. Wondering if `useServerForTests` should set up something like:

```js
context.client = got.extend({
  // ...currently existing options...

  parseJson (text: string) {
    // Unfortunately I don’t think we can get headers to switch on here, so just try both. :\
    try {
      return JSON.parse(text);
    } catch (_) {
      return ndjsonParse(text);
    }
  }
}
```

Would make other tests that need ND-JSON support nicer, too. And all our endpoints that return ND-JSON may also return regular JSON when there’s an error.

Reply:
Ha, now that I wrote that, I realize it won’t work well for ND-JSON responses with only one entry, since it’ll see them as normal JSON and you won’t get an array. 😞

Reply:
Hm, yeah. Although (and don't do this!) you could always do something like this...

```javascript
parseJson (text: string) {
  try {
    const parsed = JSON.parse(text);
    const out = [parsed];
    Object.assign(out, parsed);
  } catch (_) {
    return ndjsonParse(text);
  }
}
```
...to get the effect of both worlds, so long as the resulting parsed object doesn't have any numerical keys. 😝

Reply:
😳